src/db.coffee

```diff
@@ -1,113 +1,228 @@
-				t.string('name')
-				t.string('containerName')
-				t.string('commit')
-				t.string('imageId')
-				t.string('appId')
-				t.boolean('privileged')
-				t.json('env')
-				t.json('config')
-				t.boolean('markedForDeletion')
-		else
-			Promise.all [
-				addColumn('app', 'commit', 'string')
-				addColumn('app', 'appId', 'string')
-				addColumn('app', 'containerName', 'string')
-				addColumn('app', 'config', 'json')
-				addColumn('app', 'markedForDeletion', 'boolean')
-			]
-			.then ->
-				# When updating from older supervisors, config can be null
-				knex('app').update({ config: '{}' }).whereNull('config')
-				.then ->
-					knex('app').update({ markedForDeletion: false }).whereNull('markedForDeletion')
-
-	knex.schema.hasTable('dependentApp')
-	.then (exists) ->
-		if not exists
-			knex.schema.createTable 'dependentApp', (t) ->
-				t.increments('id').primary()
-				t.string('appId')
-				t.string('parentAppId')
-				t.string('name')
-				t.string('commit')
-				t.string('imageId')
-				t.json('config')
-				t.json('environment')
-		else
-			addColumn('dependentApp', 'environment', 'json')
-
-	knex.schema.hasTable('dependentDevice')
-	.then (exists) ->
-		if not exists
-			knex.schema.createTable 'dependentDevice', (t) ->
-				t.increments('id').primary()
-				t.string('uuid')
-				t.string('appId')
-				t.string('localId')
-				t.string('device_type')
-				t.string('logs_channel')
-				t.string('deviceId')
-				t.boolean('is_online')
-				t.string('name')
-				t.string('status')
-				t.string('download_progress')
-				t.string('is_managed_by')
-				t.dateTime('lock_expiry_date')
-				t.string('commit')
-				t.string('targetCommit')
-				t.json('environment')
-				t.json('targetEnvironment')
-				t.json('config')
-				t.json('targetConfig')
-				t.boolean('markedForDeletion')
-		else
-			Promise.all [
-				addColumn('dependentDevice', 'markedForDeletion', 'boolean')
-				addColumn('dependentDevice', 'localId', 'string')
-				addColumn('dependentDevice', 'is_managed_by', 'string')
-				addColumn('dependentDevice', 'lock_expiry_date', 'dateTime')
-		]
-])
-
-module.exports = knex
+constants = require './lib/constants'
+
+module.exports = class DB
+	constructor: ({ @databasePath } = {}) ->
+		@databasePath ?= constants.databasePath
+		@knex = Knex(
+			client: 'sqlite3'
+			connection:
+				filename: @databasePath
+			useNullAsDefault: true
+		)
+
+	addColumn: (table, column, type) =>
+		@knex.schema.hasColumn(table, column)
+		.then (exists) =>
+			if not exists
+				@knex.schema.table table, (t) ->
+					t[type](column)
+
+	dropColumn: (table, column) =>
+		@knex.schema.hasColumn(table, column)
+		.then (exists) =>
+			if exists
+				@knex.schema.table table, (t) ->
+					t.dropColumn(column)
+
+	dropTableIfExists: (tableName, trx) =>
+		knex = trx ? @knex
+		knex.schema.hasTable(tableName)
+		.then (exists) ->
+			knex.schema.dropTable(tableName) if exists
+
+	_migrateToV2: =>
+		# Drop all tables, but keep the info we need
```

Comment:
migration should happen atomically. This method will commit the tx before the new schema is in place

Reply:
Yup, this is all changing to use knex migrations